python/pyarrow/array.pxi

```diff
@@ -1646,7 +1630,6 @@ cdef class Array(_PandasConvertible):
         -------
         lst : list
         """
-        self._assert_cpu()
```

Comment:
I would expect this one to not change (not using compute kernels under the hood)?

Reply:
Handled by `__iter__` already. Isn't that what's called in the `for x in self` sugar?